build-system/tasks/extension-helpers.js

```diff
@@ -684,11 +684,14 @@ async function buildNpmBinaries(extDir, name, options) {
  * @return {!Promise}
  */
 function buildBinaries(extDir, binaries, options) {
+  // If outputPath is not defined, then use extDir
+  const {outputPath = extDir} = options;
+
   const promises = binaries.map((binary) => {
     const {babelCaller, entryPoint, external, outfile, remap, wrapper} = binary;
     const {name} = pathParse(outfile);
     const esm = argv.esm || argv.sxg || false;
-    return esbuildCompile(extDir + '/', entryPoint, `${extDir}/dist`, {
+    return esbuildCompile(extDir + '/', entryPoint, `${outputPath}/dist`, {
```

Comment:
nit: perhaps `outputPath` should be the entire directory path (consumers would pass in a path that includes `dist`)